GPU/GPUTracking/TPCClusterFinder/PeakFinder.h

```diff
@@ -14,23 +14,53 @@
 #ifndef O2_GPU_PEAK_FINDER_H
 #define O2_GPU_PEAK_FINDER_H
 
+#include "GPUGeneralKernels.h"
+#include "GPUConstantMem.h"
+
 #include "clusterFinderDefs.h"
-#include "GPUTPCClusterFinderKernels.h"
 #include "Array2D.h"
+#include "PackedCharge.h"
 
 namespace GPUCA_NAMESPACE
 {
 namespace gpu
 {
 
-class PeakFinder
+class PeakFinder : public GPUKernelTemplate
 {
 
  public:
-  static GPUd() void findPeaksImpl(int, int, int, int, GPUTPCClusterFinderKernels::GPUTPCSharedMemory&, const Array2D<gpu::PackedCharge>&, const deprecated::Digit*, uint, uchar*, Array2D<uchar>&);
+  class GPUTPCSharedMemory : public GPUKernelTemplate::GPUTPCSharedMemoryScan64<short, GPUCA_THREAD_COUNT_CLUSTERER>
```

Comment:
Does the peak finder use a scan as well, or why to we use the SharedMemoryScan class here?
So this was the place, where the thread scan count was incorrect?
Sorry, then it was my failure. I missed that part when I intregrated the stuff in O2.
(Same for deconvolution.)

Reply:
Exactly, the peak finder and deconvolutiondepend on scan. So they both gave incorrect results.

Reply:
ok, cool, so I missed that. Sorry, could have saved us some time.
Could you just clean up the naming as I commented, then we can get this merged. I't like to have it soon, to tune the performances better for AMD and NVIDIA for a better comparison. This is becoming quite urgent now.

Reply:
Should all be fixed now.

Reply:
Thx!